### ISSUE TITLE
End Time Selection Changing

### ISSUE THREAD

**Issue by alexggordon** [dangrossman/bootstrap-daterangepicker#943]
So it looks like this has been raised before in #895, but I don't think b1bc05d7 fully fixed the issue. See [this](https://jsfiddle.net/8cyw64t1/) jsfiddle with the most recent master. 

As you can see in the gif below, when you select new dates that are within the time frame of the current selection, your time selection stays the same. However, if the new start date is past the current end date, the end time is reverted to 12:00:00.

![gif](http://g.recordit.co/JX7ctrerhb.gif)

**Comment by nirmalgoswami** [comment]
same things happen to me,,when go to textbox by tab,it will change automatically to 12:00,,,,

**Comment by acl0056** [comment]
Here is a solution. Change line 856 to
```
selected = this.startDate.clone().add(selected.hour() * 60 * 60 + selected.minute() * 60 + selected.second(), "seconds");
```

**Comment by quainjn** [comment]
Yes, the issue is exactly as @alexggordon described. Choosing any start date past the current end date causes the end time to be set to the start time.

Thanks @acl0056 for the solution. I extended it a bit, since I ran into an issue where if the start time is not 00:00, the end time will get advanced by the start time. Eg. if the start time is 01:00, and the end time is 13:00, the end time will become 14:00:

```
selected = this.startDate.clone().startOf('day').add(moment.duration({'hour' : selected.hour(), 'minute': selected.minute(), 'second': selected.second()}));
if(selected.isBefore(this.startDate))
  selected = selected.add(moment.duration({'day' : 1}));
```

Essentially set the end time to the same time it previously was, either on the same day as the start day or the day after if the end time is before the start time.
